backend/src/main/java/gov/cdc/usds/simplereport/config/SecurityConfiguration.java

```diff
@@ -96,6 +96,11 @@ public IdentitySupplier getRealIdentity() {
         }
         LOG.debug("Hello JWT user {} {} ({})", firstName, lastName, email);
         return new IdentityAttributes(email, firstName, null, lastName, null);
+      } else if (principal instanceof String) {
+        String principalString = (String) principal;
+        if ("anonymousUser".equals(principalString)) {
+          return new IdentityAttributes(null, null, null, null, null);
+        }
```

Comment:
Why not check for an active patient link here? Are we too far up the filter stack for that to work yet?

Reply:
That was my impression, yeah, I think that `getRealIdentity` should get called prior to the `@PreAuthorize`, so there wouldn't yet be any data in the request context. I've made that change on a [separate branch](https://github.com/CDCgov/prime-simplereport/commit/1ad2ce1bdfd27261add87673edc398f7332277cc) and will report back my findings

Reply:
Your suspicion is correct and I've adjusted this code to pull from the `CurrentPatientContextHolder`: https://github.com/CDCgov/prime-simplereport/pull/979/commits/44e1ba11968972025ecc0846f6316db82c15276c

Reply:
I'm not entirely sure how I managed to test this before without getting 500's – my suspicion is that the back end hadn't finished restarting after deploy – 

But in fact that code in `SecurityConfiguration` _is_ too far up the filter stack to make use of `CurrentPatientContextHolder`. I have reverted these changes accordingly